### ISSUE TITLE
Can pushy drain "public queue" as well when PushManager#shutdown() is called?

### ISSUE THREAD

**Issue by TurtleShip** [relayrides/pushy#93]
Hi, @jchambers 

I understand that it is already documented that `retryQueue` will be drained when `pushManager#shutdown()` is called. However, it wasn't quite obvious to me from pushy documentation that `publicQueue` will **NOT** be drained until I wrote an integration test. 

I was wondering why pushy doesn't drain `publicQueue` along with `retryQueue.`

For now, I am doing the below when shutdown() is called.
1. check if `publicQueue` is empty.
2.  If `publicQueue` is not empty, sleep for half a second, and the go back to step 1.

I am repeating step 2 for a certain number of times just in case publicQueue never gets completely drained by pushy ( i.e. notifications in `publicQueue` gets dequeued and either gets sent or put into `retryQueue`).

Pseudo code is something as below

``` java
int currentSleepCount = 0;
while(currentSleepCount < maxSleepCount && !pushManager.getQueue().isEmpty()) {
     Thread.sleep(500);
     currentSleepCount++;
}
pushManager.shutdown();
eventLoopGroup.shutdownGracefully().await();
```

The above code is not ideal and quite hacky, but I wanted to ensure that my service processes all requests through Pushy before it is shutdown.

I thought about manually putting all notifications from `publicQueue` to `RetryQueue`, but it wasn't possible because `RetryQueue` was protected. Is there a better way to handle this?

To sum up, my questions are
1. Why does pushy not drain `publicQueue` when shutdown() is called?
2. Is there a plan to drain `publicQueue` at next iteration?
3. Is there a recommended way of draining `publicQueue` when shutdown() is called?

Thanks!
